Open Judge System/Services/OJS.Services/Data/Settings/SettingsService.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using OJS.Common.Constants;
+using OJS.Data;
+using OJS.Data.Models;
+using OJS.Services.Cache;
+
+namespace OJS.Services.Data.Settings
+{
+    public class SettingsService : ISettingsService
+    {
+        private readonly IOjsDbContext dbContext;
+        private readonly ICacheService cacheService;
+
+        public SettingsService(IOjsDbContext dbContext, ICacheService cacheService)
+        {
+            this.dbContext = dbContext;
+            this.cacheService = cacheService;
+        }
+
+        public T Get<T>(string name, T type)
+        {
+            var value = cacheService.GetOrSet(name, () =>
+            {
+                var dbSetting = this.dbContext.Settings.FirstOrDefault(s => s.Name == name);
+                if (dbSetting == null)
+                {
+                    throw new NullReferenceException();
+                }
```

Comment:
Add option for the caller to set default value, if the setting is not present. Throw only if default value is not provided.

Reply:
Done